services/AbstractService.js

```diff
@@ -0,0 +1,43 @@
+export default class AbstractService {
```

Comment:
First of all I strongly agree that composition is better than abstraction, but we have to remember that AbstractService will be only one in the memory instead of multiple version of the same (this happens if we would use mixins) that is why it's declared as a class. 
Of course you got right that there is perfectly fine library `qs` for query strings but I do not want to include all library in the bundle. I encourage you to think about it one more time. We should wisely choose what should be included in bundle.